File: Backend/src/main/java/com/luckyseven/backend/domain/email/repository/EmailVerificationTokenRepository.java
package com.luckyseven.backend.domain.email.repository;


import com.luckyseven.backend.domain.email.entity.EmailVerificationToken;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface EmailVerificationTokenRepository extends JpaRepository<EmailVerificationToken,Long> {
  Optional<EmailVerificationToken> findByEmail(String email);

  Optional<EmailVerificationToken> findByToken(String token);

  void deleteByEmail(String email);
}

File: Backend/src/main/kotlin/com/luckyseven/backend/domain/email/repository/EmailVerificationTokenRepository.kt
package com.luckyseven.backend.domain.email.repository

import com.luckyseven.backend.domain.email.entity.EmailVerificationToken
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface EmailVerificationTokenRepository : JpaRepository<EmailVerificationToken, Long> {
    fun findByToken(token: String): EmailVerificationToken?
    
    fun deleteByEmail(email: String)
} 